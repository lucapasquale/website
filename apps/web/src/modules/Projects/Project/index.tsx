import React from 'react'

import { TechnologyTag } from '~common/components/TechnologyTag'

import type { Project as ProjectType } from '../logic/getProjects'

import { Header } from './Header'

type Props = {
  project: ProjectType
}

export function Project({ project }: Props) {
  return (
    <article className="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch">
      <div className="min-w-[275px] max-w-[275px] border border-gray-500 text-[0px]">
        {project.image}
      </div>

      <div className="mt-4 ml-0 sm:ml-8 max-w-md">
        <Header project={project} />

        <ul className="flex flex-row flex-wrap gap-2 mb-4">
          {project.technologies.map((technology, index) => (
            <li key={index}>
              <TechnologyTag technology={technology} />
            </li>
          ))}
        </ul>

        <p>{project.description}</p>
      </div>
    </article>
  )
}
