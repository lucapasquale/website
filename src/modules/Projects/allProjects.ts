import { Technology } from "~components/TechnologyTag";

export type Project = {
  title: string;
  description: string;
  url: string;
  githubUrl?: string;
  imagePath: string;
  technologies: Technology[];
};

export const ALL_PROJECTS: Project[] = [
  {
    title: "Watchlist",
    url: "https://watchlist.luca.codes",
    githubUrl: "https://github.com/lucapasquale/watchlist/",
    imagePath: "/images/projects/watchlist.webp",
    description: "Create and share your favorite videos from YouTube, Twitch and Twitter",
    technologies: ["typescript", "react", "vite", "tailwind", "nestjs"],
  },
  {
    title: "Botchini",
    url: "https://botchini.luca.codes",
    githubUrl: "https://github.com/lucapasquale/botchini/",
    imagePath: "/images/projects/botchini.webp",
    description:
      "Discord bot that notifies when a creator starts streaming on Twitch or uploads a video to YouTube",
    technologies: ["elixir", "postgresql"],
  },
  {
    title: "Up We Go",
    url: "https://play.google.com/store/apps/details?id=com.Bendev.UpWeGo",
    imagePath: "/images/projects/up-we-go.webp",
    description:
      "Climb to the top in this arcade game, try different characters and compete with your friends for the highest score!",
    technologies: ["unity", "csharp"],
  },
  {
    title: "Portfolio",
    url: "https://www.luca.codes",
    githubUrl: "https://github.com/lucapasquale/website",
    imagePath: "/images/projects/website.webp",
    description: "You know, the one that has this text and is right in front of you.",
    technologies: ["typescript", "react", "vite", "tailwind"],
  },
  {
    title: "Linking Bio",
    url: "https://github.com/lucapasquale/linking-bio",
    githubUrl: "https://github.com/lucapasquale/linking-bio",
    imagePath: "/images/projects/linking-bio.webp",
    description:
      "Share more than a single link in your social media. Create your own page, where you can customize and have as many links as you would like.",
    technologies: ["nestjs", "graphql", "postgresql", "typescript", "react", "apollo", "nextjs"],
  },
  {
    title: "Fit It",
    url: "https://github.com/lucapasquale/FitIt",
    githubUrl: "https://github.com/lucapasquale/FitIt",
    imagePath: "/images/projects/fit-it.webp",
    description:
      "Solve hundreds of different puzzles in this great mobile game. Game was published on Android Play Store, where it had over a thousand downloads.",
    technologies: ["unity", "csharp"],
  },
];
