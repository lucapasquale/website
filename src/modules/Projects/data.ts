import { Props as ProjectProps } from './Project'

export const projects: ProjectProps[] = [
  {
    title: 'Linking Bio',
    url: 'https://www.linkingbio.com',
    imageSrc: '/img/linking-bio.png',
    description: `Share more than a single link in your social media.
      Create your own page, where you can customize and have as many links as you would like.`,
    technologies: [
      {
        title: 'Front-end:',
        list: ['React', 'Apollo Client', 'Emotion', 'TypeScript'],
      },
      {
        title: 'Back-end:',
        list: ['GraphQL', 'DynamoDB', 'Serverless', 'TypeScript'],
      },
    ],
  },

  {
    title: 'My portfolio webpage',
    url: 'https://github.com/lucapasquale/website',
    imageSrc: '/img/website.png',
    description: `You know, the one that has this text and is right in front of you.`,
    technologies: [
      {
        title: 'Technologies:',
        list: ['React', 'Styled-Components', 'Parcel Bundler', 'TypeScript'],
      },
    ],
  },

  {
    title: 'Fit It',
    url: 'https://github.com/lucapasquale/FitIt',
    imageSrc: '/img/fit-it.png',
    description: `Solve hundreds of different puzzles in this great mobile game.
      Game was published on Android Play Store, where it had over a thousand downloads.`,
    technologies: [
      {
        title: 'Technologies:',
        list: ['Unity Game Engine', 'Android', 'iOS', 'C#'],
      },
    ],
  },
]
