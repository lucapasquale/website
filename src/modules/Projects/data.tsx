import Image from 'next/image'
import { Props as ProjectProps } from './Project'

export function getProjects(isDarkTheme: boolean): ProjectProps[] {
  return [
    {
      title: 'Botchini',
      url: 'https://github.com/lucapasquale/botchini/',
      image: (
        <Image
          placeholder="blur"
          alt="botchini project image"
          src={require('../../../public/images/projects/botchini.webp')}
        />
      ),
      description: `Discord bot to notify when a Twitch streamer starts streaming`,
      technologies: [
        {
          title: 'Technologies:',
          list: ['Elixir', 'PostgreSQL'],
        },
      ],
    },

    {
      title: 'Up We Go',
      url: 'https://play.google.com/store/apps/details?id=com.Bendev.UpWeGo',
      image: (
        <Image
          placeholder="blur"
          alt="Up We Go project image"
          src={require('../../../public/images/projects/up-we-go.webp')}
        />
      ),
      description: `Climb to the top in this arcade game, try different characters
        and compete with your friends for the highest score!`,
      technologies: [
        {
          title: 'Technologies:',
          list: ['Unity Game Engine', 'Android', 'iOS', 'C#'],
        },
      ],
    },

    {
      title: 'Linking Bio',
      url: 'https://www.linkingbio.com',
      image: (
        <Image
          placeholder="blur"
          alt="Linking bio project image"
          src={require('../../../public/images/projects/linking-bio.webp')}
        />
      ),
      description: `Share more than a single link in your social media.
        Create your own page, where you can customize and have as many links as you would like.`,
      technologies: [
        {
          title: 'Back-end:',
          list: ['GraphQL', 'NestJS', 'PostgreSQL', 'TypeScript'],
        },
        {
          title: 'Front-end:',
          list: ['Next.js', 'Apollo Client', 'Material-UI', 'TypeScript'],
        },
      ],
    },

    {
      title: 'My portfolio webpage',
      url: 'https://github.com/lucapasquale/website',
      image: isDarkTheme ? (
        <Image
          placeholder="blur"
          alt="Website project image"
          src={require('../../../public/images/projects/website-light.webp')}
        />
      ) : (
        <Image
          placeholder="blur"
          alt="Website project image"
          src={require('../../../public/images/projects/website-dark.webp')}
        />
      ),
      description: `You know, the one that has this text and is right in front of you.`,
      technologies: [
        {
          title: 'Technologies:',
          list: ['React', 'Next.js', 'Styled-Components', 'TypeScript'],
        },
      ],
    },

    {
      title: 'Fit It',
      url: 'https://github.com/lucapasquale/FitIt',
      image: (
        <Image
          placeholder="blur"
          alt="Fit It project image"
          src={require('../../../public/images/projects/fit-it.webp')}
        />
      ),
      description: `Solve hundreds of different puzzles in this great mobile game.
        Game was published on Android Play Store, where it had over a thousand downloads.`,
      technologies: [
        {
          title: 'Technologies:',
          list: ['Unity Game Engine', 'Android', 'iOS', 'C#'],
        },
      ],
    },
  ]
}
