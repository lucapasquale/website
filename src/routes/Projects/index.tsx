import React, { FC } from 'react'
import Helmet from 'react-helmet'
import styled from 'styled-components'

import { Hero } from '../../components/Hero'
import { Layout } from '../../components/Layout'
import { Project } from './components/Project'

const Wrapper = styled.div({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',

  padding: '32px',
})

const Divider = styled.hr(({ theme }) => ({
  width: '100%',
  maxWidth: '500px',
  margin: '32px 0px',

  borderBottomColor: theme.colors.primary,
  borderBottomWidth: '0.5px',
}))

export const Projects: FC = () => (
  <>
    <Helmet>
      <title>Luca Pasquale - Projects</title>
      <link rel="canonical" href="https://lucapasquale.github.io/projects" />
    </Helmet>

    <Layout>
      <Hero
        title="Projects"
        subTitle="Some things I've created over the years"
      />

      <Wrapper>
        <Project
          title="Linking Bio"
          url="https://www.linkingbio.com"
          imageSrc={require('../../assets/img/linking-bio.png')}
          description="Share more than a single link in your social media. Create your own page, where you can customize and have as many links as you would like."
          technologies={[
            {
              title: 'Front-end',
              list: ['React', 'Apollo Client', 'Emotion', 'TypeScript'],
            },
            {
              title: 'Back-end',
              list: ['GraphQL', 'DynamoDB', 'Serverless', 'TypeScript'],
            },
          ]}
        />

        <Divider />
      </Wrapper>
    </Layout>
  </>
)
