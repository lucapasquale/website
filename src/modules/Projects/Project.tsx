import { ExternalLinkIcon, GitHubLogoIcon } from "@radix-ui/react-icons";

import { Link } from "~components/Link";
import { TechnologyTag } from "~components/TechnologyTag";

import { type Project } from "./allProjects";

type Props = {
  project: Project;
  preload?: boolean;
};

export function ProjectCard({ project, preload = false }: Props) {
  return (
    <article className="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch">
      <div className="w-[275px] border-0 border-b border-solid border-gray-500">
        <img rel={preload ? "preload" : undefined} src={project.imagePath} alt={project.title} />
      </div>

      <div className="mt-4 ml-0 sm:ml-8 max-w-md">
        <Header project={project} />

        <ul className="flex flex-row flex-wrap gap-2 mb-4">
          {project.technologies.map((technology, index) => (
            <li key={index}>
              <TechnologyTag technology={technology} />
            </li>
          ))}
        </ul>

        <p>{project.description}</p>
      </div>
    </article>
  );
}

function Header({ project }: Props) {
  return (
    <div className="w-fit mb-4">
      <Link className="flex items-center gap-2 text-foreground mb-2 sm:mb-0" to={project.url}>
        <h3 className="text-2xl">{project.title}</h3>

        <ExternalLinkIcon className="w-6 h-6" />
      </Link>

      {project.githubUrl && (
        <Link className="flex items-center gap-1 text-foreground" to={project.githubUrl}>
          <GitHubLogoIcon />

          <h4 className="text-md">GitHub</h4>
        </Link>
      )}
    </div>
  );
}
