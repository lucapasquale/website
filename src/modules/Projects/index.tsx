import { Hero } from "~components/Hero";
import { Separator } from "~components/Separator";

import { ALL_PROJECTS } from "./allProjects";
import { ProjectCard } from "./Project";

export function Page() {
  return (
    <>
      <Hero title="Projects" subTitle="Some things I've created over the years" />

      <div className="flex flex-col items-center justify-between">
        {ALL_PROJECTS.map((project, idx) => (
          <section>
            <ProjectCard key={idx} project={project} preload={idx <= 2} />

            {idx < ALL_PROJECTS.length - 1 && <Separator className="block sm:hidden" />}
          </section>
        ))}
      </div>
    </>
  );
}
