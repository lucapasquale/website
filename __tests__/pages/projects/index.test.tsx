import { render, screen, within } from '@testing-library/react'

import Projects from '../../../src/pages/projects'

describe('Projects', () => {
  it('snapshot', () => {
    const { container } = render(<Projects />)

    expect(container).toMatchSnapshot()
  })

  it('renders the page', async () => {
    render(<Projects />)

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()

    const projects = await screen.findAllByRole('article')

    projects.forEach((project) => {
      expect(within(project).getByRole('img')).toBeInTheDocument()
      expect(within(project).getByRole('heading', { level: 3 })).toBeInTheDocument()
    })
  })
})
