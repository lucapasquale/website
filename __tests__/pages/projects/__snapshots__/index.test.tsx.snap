// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects snapshot 1`] = `
<div>
  <header
    class="mb-12"
  >
    <h1
      class="text-4xl font-bold pb-2"
    >
      Projects
    </h1>
    <h2
      class="text-gray-400"
    >
      Some things I've created over the years
    </h2>
  </header>
  <div
    class="flex flex-col items-center justify-between"
  >
    <article
      class="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch"
    >
      <div
        class="min-w-[275px] max-w-[275px] border border-gray-500 text-[0px]"
      >
        <img
          alt="test-img"
        />
      </div>
      <div
        class="mt-4 ml-0 sm:ml-8 max-w-md"
      >
        <div
          class="mb-4"
        >
          <a
            class="flex items-baseline gap-2 text-gray-300 mb-2 sm:mb-0"
            href="https://github.com/lucapasquale/botchini/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h3
              class="text-2xl"
            >
              Botchini
            </h3>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
              />
            </svg>
          </a>
          <a
            class="flex items-center gap-1 text-gray-300"
            href="https://github.com/lucapasquale/botchini/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
            <h4
              class="text-md"
            >
              GitHub
            </h4>
          </a>
        </div>
        <ul
          class="flex flex-row flex-wrap gap-2 mb-4"
        >
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#6e4a7e]"
          >
            elixir
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#0064a5]"
          >
            postgresql
          </li>
        </ul>
        <p>
          Discord bot to notify when a Twitch streamer starts streaming
        </p>
      </div>
    </article>
    <article
      class="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch"
    >
      <div
        class="min-w-[275px] max-w-[275px] border border-gray-500 text-[0px]"
      >
        <img
          alt="test-img"
        />
      </div>
      <div
        class="mt-4 ml-0 sm:ml-8 max-w-md"
      >
        <div
          class="mb-4"
        >
          <a
            class="flex items-baseline gap-2 text-gray-300 mb-2 sm:mb-0"
            href="https://play.google.com/store/apps/details?id=com.Bendev.UpWeGo"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h3
              class="text-2xl"
            >
              Up We Go
            </h3>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
              />
            </svg>
          </a>
        </div>
        <ul
          class="flex flex-row flex-wrap gap-2 mb-4"
        >
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#222c37]"
          >
            unity
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#fbfcff] bg-[#178600]"
          >
            c#
          </li>
        </ul>
        <p>
          Climb to the top in this arcade game, try different characters
      and compete with your friends for the highest score!
        </p>
      </div>
    </article>
    <article
      class="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch"
    >
      <div
        class="min-w-[275px] max-w-[275px] border border-gray-500 text-[0px]"
      >
        <img
          alt="test-img"
        />
      </div>
      <div
        class="mt-4 ml-0 sm:ml-8 max-w-md"
      >
        <div
          class="mb-4"
        >
          <a
            class="flex items-baseline gap-2 text-gray-300 mb-2 sm:mb-0"
            href="https://www.linkingbio.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h3
              class="text-2xl"
            >
              Linking Bio
            </h3>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
              />
            </svg>
          </a>
        </div>
        <ul
          class="flex flex-row flex-wrap gap-2 mb-4"
        >
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#fdfeff] bg-[#e0234e]"
          >
            nestjs
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#030303] bg-[#e10098]"
          >
            graphql
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#0064a5]"
          >
            postgresql
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#edeff7] bg-[#2b7489]"
          >
            typescript
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-800 bg-[#61dafb]"
          >
            react
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#eff1f5] bg-[#7156d9]"
          >
            apollo
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#111]"
          >
            next.js
          </li>
        </ul>
        <p>
          Share more than a single link in your social media.
      Create your own page, where you can customize and have as many links as you would like.
        </p>
      </div>
    </article>
    <article
      class="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch"
    >
      <div
        class="min-w-[275px] max-w-[275px] border border-gray-500 text-[0px]"
      >
        <img
          alt="test-img"
        />
      </div>
      <div
        class="mt-4 ml-0 sm:ml-8 max-w-md"
      >
        <div
          class="mb-4"
        >
          <a
            class="flex items-baseline gap-2 text-gray-300 mb-2 sm:mb-0"
            href="https://lucapasquale.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h3
              class="text-2xl"
            >
              Portfolio
            </h3>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
              />
            </svg>
          </a>
          <a
            class="flex items-center gap-1 text-gray-300"
            href="https://github.com/lucapasquale/website"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
            <h4
              class="text-md"
            >
              GitHub
            </h4>
          </a>
        </div>
        <ul
          class="flex flex-row flex-wrap gap-2 mb-4"
        >
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#edeff7] bg-[#2b7489]"
          >
            typescript
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-800 bg-[#61dafb]"
          >
            react
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#111]"
          >
            next.js
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-800 bg-[#06b6d4]"
          >
            tailwind
          </li>
        </ul>
        <p>
          You know, the one that has this text and is right in front of you.
        </p>
      </div>
    </article>
    <article
      class="mt-8 mb-10 flex w-full flex-col sm:flex-row items-center sm:items-stretch"
    >
      <div
        class="min-w-[275px] max-w-[275px] border border-gray-500 text-[0px]"
      >
        <img
          alt="test-img"
        />
      </div>
      <div
        class="mt-4 ml-0 sm:ml-8 max-w-md"
      >
        <div
          class="mb-4"
        >
          <a
            class="flex items-baseline gap-2 text-gray-300 mb-2 sm:mb-0"
            href="https://github.com/lucapasquale/FitIt"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h3
              class="text-2xl"
            >
              Fit It
            </h3>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
              />
            </svg>
          </a>
          <a
            class="flex items-center gap-1 text-gray-300"
            href="https://github.com/lucapasquale/FitIt"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
            <h4
              class="text-md"
            >
              GitHub
            </h4>
          </a>
        </div>
        <ul
          class="flex flex-row flex-wrap gap-2 mb-4"
        >
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-gray-200 bg-[#222c37]"
          >
            unity
          </li>
          <li
            class="inline-block py-0.5 px-2 rounded-sm text-sm uppercase font-bold text-[#fbfcff] bg-[#178600]"
          >
            c#
          </li>
        </ul>
        <p>
          Solve hundreds of different puzzles in this great mobile game.
      Game was published on Android Play Store, where it had over a thousand downloads.
        </p>
      </div>
    </article>
  </div>
</div>
`;
